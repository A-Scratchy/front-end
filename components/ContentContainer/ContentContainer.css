.ContentContainer {
  align-items: center;
  background-size: cover;
  display: flex;
  flex-direction: column;
  min-height: 250px;
  width: 100%;
}

/* Dark Themes */
.secondary {
  background-color: var(--secondary);
  color: var(--white);
}

/* Light Themes */
.gray,
.white {
  color: var(--secondary);
}

.gray {
  background-color: var(--gray);
}

.white {
  background-color: var(--white);
}
