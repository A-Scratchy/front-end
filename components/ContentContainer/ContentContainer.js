import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './ContentContainer.css';

ContentContainer.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  backgroundImageSource: PropTypes.string,
  theme: PropTypes.oneOf(['gray', 'secondary', 'white']),
};

ContentContainer.defaultProps = {
  className: undefined,
  backgroundImageSource: undefined,
  children: undefined,
  theme: 'gray',
};

function ContentContainer({ children, className, backgroundImageSource, theme }) {
  return (
    <div
      backgroundImageSource={backgroundImageSource}
      className={classNames(className, styles.ContentContainer, styles[theme])}
    >
      {children}
    </div>
  );
}

export default ContentContainer;
