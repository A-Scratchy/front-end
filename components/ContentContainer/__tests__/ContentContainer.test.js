/* eslint-env jest */
import React from 'react';
import createSnapshotTest from 'test-utils/createSnapshotTest';

import ContentContainer from '../ContentContainer';

describe('ContentContainer', () => {
  it('should render with many props assigned', () => {
    createSnapshotTest(
      <ContentContainer className="test-class" backgroundImageSource="testImage" theme="gray">
        Test Children
      </ContentContainer>,
    );
  });

  it('should render common use case', () => {
    createSnapshotTest(
      <ContentContainer className="test-class" theme="secondary">
        Test Children 2
      </ContentContainer>,
    );
  });
});
