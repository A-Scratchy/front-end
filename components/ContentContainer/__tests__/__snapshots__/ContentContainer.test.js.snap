// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentContainer should render common use case 1`] = `
<div
  className="test-class ContentContainer secondary"
>
  Test Children 2
</div>
`;

exports[`ContentContainer should render with many props assigned 1`] = `
<div
  backgroundImageSource="testImage"
  className="test-class ContentContainer gray"
>
  Test Children
</div>
`;
